File: answers/leetcode/Plus One/Plus One.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        for (int i = digits.length - 1; i >= 0; i--) {
            digits[i] += carry;
            carry = digits[i] / 10;
            digits[i] %= 10;
        }
        if (carry > 0) {
            int[] number = new int[digits.length + 1];
            number[0] = carry;
            for (int i = 0; i < digits.length; i++) {
                number[i + 1] = digits[i];
            }
            return number;
        } else {
            return digits;
        }
    }
}

File: answers/leetcode/Plus One/Plus One.cpp
//@result 108 / 108 test cases passed. Status: Accepted Runtime: 4 ms Submitted: 0 minutes ago You are here! Your runtime beats 5.42% of cpp submissions.

class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
		int carry = 1;
		int pos = digits.size() - 1;
		while (carry && pos >= 0)
		{
			digits[pos] += carry;
			carry = digits[pos] / 10;
			digits[pos] %= 10;
			-- pos;
		}
		if (carry)
		{
			digits.insert(digits.begin(), carry);
		}
		return digits;
    }
};
